### PR TITLE
Update zbus deps

### PR DESCRIPTION
**Pull request by flukejones**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,8 +24,8 @@ include = [
 dbus = { version = "0.9", optional = true }
 lazy_static = { version = "1", optional = true }
 image = { version = "0.24", optional = true }
-zvariant = { version = "3.2", optional = true }
-zvariant_derive = { version = "3.2", optional = true }
+zvariant = { version = "3.7", optional = true }
+zvariant_derive = { version = "3.7", optional = true }
```
Comment (zeenix) [inline Cargo.toml:28]: @flukejones Hey. Glad to see you doing the great work here. :) Is the direct dependency on zvariant crates really needed here btw? It's best to use it through the zbus re-export. Also, `zvariant_derive` should almost never be used directly. The macros are re-exported by `zvariant`.

File: Cargo.toml [hunk 2]
```diff
@@ -24,9 +24,9 @@ include = [
 dbus = { version = "0.9", optional = true }
 lazy_static = { version = "1", optional = true }
 image = { version = "0.24", optional = true }
-zvariant = { version = "3", optional = true }
-zvariant_derive = { version = "3", optional = true }
-zbus = { version = "2", optional = true }
+zvariant = { version = "3.7", optional = true }
+zvariant_derive = { version = "3.7", optional = true }
+zbus = { version = "3.4", optional = true }
```
Comment (zeenix) [inline Cargo.toml:29]: We are already at 3.5 and 3.6 hopefully coming this week.

**Comment by hoodie** [comment]
thanks for the work, but this way to much in one commit.
1. please don't do 5 things in one commit, the PR heading did not mention anything about error propagation, I was genuinely surprised why we need a third "update zbus PR"
2. one does not simply reformat everything in a commit that says "bump version...", this makes it quite hard to review where there were actual changes
3. please adhere to conventional commits

**Comment by hoodie** [comment]
Ok, so I just did some homework:

1. notify-rust is now formatted
   I don't love some of it, but it is contribute-friendlier now, you shouldn't have to be afraid of running `cargo fmt`
2. I updated zbus to 3.5
   marking it as a `feat` so that users don't complain again
3. I added a ci check for commit messages
   hoping this will be a bit helpful in communicating the odd conventional-commit requirement
4. **I took the liberty of rebasing your PR** and reduce it to the most interesting change: the error handling
   unfortunately I was not able to push on your branch (that some times works not this time 🤷)
   instead I push that branch to [`feature/flukejones`](https://github.com/hoodie/notify-rust/tree/feature/flukejones)
   please have a look

Thanks for your help, despite my apparent gnarling it is still very much appreciated!!!

**Comment by flukejones** [comment]
@hoodie I'll close this PR for a new one. The most important thing for me was to update zbus so I wasn't including a duplicate older crate in builds.

Regarding the error propagation, I think this is also an important thing as you can potentially crash a thread or program when the errors could be caught and handled.

**Comment by zeenix** [comment]
> I'll close this PR for a new one.

Just curious, why close the PR? :thinking:  Your new PR does the same.

**Comment by flukejones** [comment]
> > I'll close this PR for a new one.
> 
> Just curious, why close the PR? thinking Your new PR does the same.

Bunch of conflicts due to separate dep updates, and I couldn't do a rebase on the existing branch because it will break my dependency chain in asusctl (which is using my fork of notify for now).

**Comment by zeenix** [comment]
> > Just curious, why close the PR? thinking Your new PR does the same.
> 
> Bunch of conflicts due to separate dep updates, and I couldn't do a rebase on the existing branch because it will break my dependency chain in asusctl (which is using my fork of notify for now).

Thanks for explaining. Ah, I see now that you submitted your main branch for the PR. That's one reason I try to always use separate branches. Avoids the churn. :)

**Pull request closed without merging**